src/lib.rs

```diff
@@ -51,9 +57,511 @@ mod writer;
+            "message" => Ok(MediaType::Message),
+            "image" => Ok(MediaType::Image),
+            _ => Err(ParseEnumError::Invalid(s.to_string())),
+        }
+    }
+}
+
+impl Display for MediaType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
+
+/// Transport Protocol for the media
+///
+/// See [RFC 8866 Section 5.14](https://datatracker.ietf.org/doc/html/rfc8866#section-5.14)
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum TransportProto {
+    /// Direct UDP
+    Udp,
+    /// RTP over UDP
+    RtpAvp,
+    /// Secure RTP over UDP
+    RtpSavp,
+    /// Secure RTP over UDP with RTCP-based feedback
+    RtpSavpf,
+}
+
+impl TransportProto {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            // The strings are case-insensitive, but the spec (RFC 8866) uses lower-case of the "udp" protocol
+            // and upper-case for the others so keeping it the same
+            TransportProto::Udp => "udp",
+            TransportProto::RtpAvp => "RTP/AVP",
+            TransportProto::RtpSavp => "RTP/SAVP",
+            TransportProto::RtpSavpf => "RTP/SAVPF",
+        }
+    }
+}
+
+impl FromStr for TransportProto {
+    type Err = ParseEnumError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // The strings are case-insensitive, but the spec (RFC 8866) uses lower-case of the "udp" protocol
+        // and upper-case for the others so keeping it the same
+        if "udp".eq_ignore_ascii_case(s) {
+            Ok(TransportProto::Udp)
+        } else if "RTP/AVP".eq_ignore_ascii_case(s) {
+            Ok(TransportProto::RtpAvp)
+        } else if "RTP/SAVP".eq_ignore_ascii_case(s) {
+            Ok(TransportProto::RtpSavp)
+        } else if "RTP/SAVPF".eq_ignore_ascii_case(s) {
+            Ok(TransportProto::RtpSavpf)
+        } else {
+            Err(ParseEnumError::Invalid(s.to_string()))
+        }
+    }
+}
+
+impl Display for TransportProto {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
+
+/// RtpMap Attribute
+///
+/// See [RFC 8866 Section 6.6](https://datatracker.ietf.org/doc/html/rfc8866#section-6.6) for more details
+#[derive(Debug, Clone, PartialEq)]
+pub struct RtpMap {
+    /// Payload type, a numerical value between 0 and 127
+    pub pt: u8,
+    /// Name of the encoding
+    // TODO: is it useful to have an enum for all known encoding?
+    pub encoding: String,
+    /// Clock rate
+    pub clock_rate: u32,
+    /// Encoding parameters. Currently used only for audio channel count
+    pub params: Option<String>,
+}
+
+impl RtpMap {
+    /// Converts the RtpMap to String
+    pub fn as_string(&self) -> String {
+        let mut s = format!("{} {}/{}", self.pt, self.encoding, self.clock_rate);
+        if let Some(params) = &self.params {
+            s += format!("/{params}").as_str();
+        }
+        s
+    }
```

Comment:
This should probably become the `Display` implementation and then you can just do `to_string()` to get this